src/common/mfu_util.c

```diff
@@ -1105,6 +1109,284 @@ bool mfu_is_lustre(const char* path)
+    }
+}
+
+int
+mfu_mount_info_parse_fstype_line(char                  *line,
+                                 size_t                 size,
+                                 struct mfu_mount_info *mip)
+{
+    char *found   = NULL;
+    char *saveptr = NULL;
+
+    found = strtok_r(line, ".", &saveptr);
+    if (found == NULL) {
+        return(ENOENT);
+    }
+
+    mip->mfu_mi_fstype = malloc(strlen(found)+1);
+    snprintf(mip->mfu_mi_fstype, strlen(found)+1, "%s", found);
+
+    found = strtok_r(NULL, ".", &saveptr);
+    if (found == NULL) {
+        return(0);
+    }
+
+    mip->mfu_mi_fssubtype = malloc(strlen(found)+1);
+    snprintf(mip->mfu_mi_fssubtype, strlen(found)+1, "%s", found);
+    return(0);
+}
+
+int
+mfu_mount_info_parse_mount_line(char                  *line,
+                                size_t                 size,
+                                struct mfu_mount_info *mip)
+{
+    char *found   = NULL;
+    char *saveptr = NULL;
+    int   num     = 0;
+
+    found = strtok_r(line, " ", &saveptr);
+    while (found != NULL) {
+        num++;
+        switch (num) {
+            case 1:
+                mip->mfu_mi_source = malloc(strlen(found)+1);
+                snprintf(mip->mfu_mi_source, strlen(found)+1, "%s", found);
+                break;
+            case 2:
+                mip->mfu_mi_mountpoint = malloc(strlen(found)+1);
+                snprintf(mip->mfu_mi_mountpoint, strlen(found)+1, "%s", found);
+                break;
+            case 3:
+                mfu_mount_info_parse_fstype_line(found, strlen(found)+1, mip);
+                break;
+            case 4:
+                mip->mfu_mi_mountopts = malloc(strlen(found)+1);
+                snprintf(mip->mfu_mi_mountopts, strlen(found)+1, "%s", found);
+                break;
+            default:
+                break;
+        }
+
+        found = strtok_r(NULL, " ", &saveptr);
+    }
+
+    return(0);
+}
+
+int
+mfu_mount_info_get_filesystem_list(struct list_head *mfu_mi_list)
+{
+    size_t                 len     = 0;
+    ssize_t                bytes   = 0;
+    char                  *line    = NULL;
+    struct mfu_mount_info *itemp   = NULL;
+    FILE                  *mountfp = NULL;
+    int                    rc      = 0;
+    struct stat            st;
+
+    if ( mfu_mi_list == NULL ) {
+        return(EINVAL);
+    }
+
+    errno = 0;
+    mountfp = fopen("/proc/self/mounts", "r");
+    if (mountfp == NULL) {
+        return(errno);
+    }
+
+    bytes = getline(&line, &len, mountfp);
+    while (bytes != -1)
+    {
+        itemp = malloc(sizeof(struct mfu_mount_info));
+        if (itemp == NULL) {
+            fclose(mountfp);
+            return(ENOMEM);
+        }
+        memset(itemp, 0, sizeof(struct mfu_mount_info));
+        INIT_LIST_HEAD(&itemp->mfu_mi_linkage);
+
+        /*
+         * Parse one of the lines in /proc/self/mounts
+         */
+        mfu_mount_info_parse_mount_line(line, len, itemp);
```

Comment:
Let's handle any parse error here, in case someone has a bad line in their file.  In particular, I think we should avoid calling the subsequent ``stat()`` if we failed to parse the line and maybe skip adding an entry in the list.

Reply:
My initial goal here was to try and be as lenient as possible. Basically just wanted to have a simple data structure that included everything in "/proc/self/mounts". If that possibly had a bad entry, then I was okay with it being reflected in the list. I am totally fine with whatever you prefer here for handling parse errors.